collectors/charts.d.plugin/speedtest/speedtest.chart.sh

```diff
@@ -0,0 +1,160 @@
+
+# netdata
+# real-time performance and health monitoring, done right!
+# (C) 2016 Costa Tsaousis <costa@tsaousis.gr>
+#
+
+# if this chart is called X.chart.sh, then all functions and global variables
+# must start with X_
+
+# _update_every is a special variable - it holds the number of seconds
+# between the calls of the _update() function
+speedtest_update_every=120
+
+# the priority is used to sort the charts on the dashboard
+# 1 = the first chart
+speedtest_priority=150000
+
+# to enable this chart, you have to set this to 12345
+# (just a demonstration for something that needs to be checked)
+speedtest_magic_number=
+
+# global variables to store our collected data
+# remember: they need to start with the module name speedtest_
+speedtest_download=0   
+speedtest_upload=0
+speedtest_download_bytes=0
+speedtest_upload_bytes=0
+speedtest_idle_latency=0
+speedtest_download_latency=0
+speedtest_upload_latency=0
+speedtest_idle_jitter=0
+speedtest_download_jitter=0
+speedtest_upload_jitter=0
+speedtest_packetloss=0
+speedtest_last=0
+speedtest_count=0
+
+speedtest_get() {
+  # do all the work to collect / calculate the values
+  # for each dimension
+  #
+  # Remember:
+  # 1. KEEP IT SIMPLE AND SHORT
+  # 2. AVOID FORKS (avoid piping commands)
+  # 3. AVOID CALLING TOO MANY EXTERNAL PROGRAMS
+  # 4. USE LOCAL VARIABLES (global variables may overlap with other modules)
+
+  output=$(speedtest --format=csv)
+  
+  speedtest_download=$(echo $output | awk -F, '{print $6}' | tr -d '"')
+  speedtest_upload=$(echo $output | awk -F, '{print $7}' | tr -d '"')
+  speedtest_download_bytes=$(echo $output | awk -F, '{print $8}' | tr -d '"')
+  speedtest_upload_bytes=$(echo $output | awk -F, '{print $9}' | tr -d '"')
+  speedtest_idle_latency=$(echo $output | awk -F, '{print $3}' | tr -d '"')
+  speedtest_download_latency=$(echo $output | awk -F, '{print $12}' | tr -d '"')
+  speedtest_upload_latency=$(echo $output | awk -F, '{print $16}' | tr -d '"')
+  speedtest_idle_jitter=$(echo $output | awk -F, '{print $4}' | tr -d '"')
+  speedtest_download_jitter=$(echo $output | awk -F, '{print $13}' | tr -d '"')
+  speedtest_upload_jitter=$(echo $output | awk -F, '{print $17}' | tr -d '"')
+  speedtest_packetloss=$(echo $output | awk -F, '{print $5}' | tr -d '"')
+  # this should return:
+  #  - 0 to send the data to netdata
+  #  - 1 to report a failure to collect the data
+
+  return 0
+}
+
+# _check is called once, to find out if this chart should be enabled or not
+speedtest_check() {
+  # this should return:
+  #  - 0 to enable the chart
+  #  - 1 to disable the chart
+
+  # check something
+  
+  # check that we can collect data
+  speedtest_get || return 1
+
+  return 0
+}
+
+# _create is called once, to create the charts
+speedtest_create() {
+  cat << EOF
+CHART speedtest.download '' 'Download Bandwidth' 'bps' 'bandwidth' 'speedtest.download' area $((speedtest_priority)) $speedtest_update_every '' '' 'speedtest'
```

Comment:
I suggest to also inform [plugin](https://learn.netdata.cloud/docs/agent/collectors/plugins.d#chart) name here. We know it is a charts.d.plugin.

Reply:
@thiagoftsm is there an example for this?